### PR TITLE
CA-273674 Added warning when shutting down pool master

### PR DESCRIPTION
**Pull request by serencorbett1**
Signed-off-by: serenc <seren.corbett@citrix.com>

### CODE REVIEW

File: XenAdmin/Commands/ShutDownHostCommand.cs [hunk 1]
```diff
@@ -143,21 +145,40 @@ protected override string ConfirmationDialogText
...
-
-                if (hasRunningVMs)
-                    return hosts.Count == 1
-                        ? string.Format(Messages.CONFIRM_SHUTDOWN_SERVER, hosts[0].Name())
-                        : Messages.CONFIRM_SHUTDOWN_SERVERS;
-
-                return hosts.Count == 1
-                    ? string.Format(Messages.CONFIRM_SHUTDOWN_SERVER_NO_VMS, hosts[0].Name())
-                    : Messages.CONFIRM_SHUTDOWN_SERVERS_NO_VMS;
+                    sb = hciHosts.Count == 1
```
Comment (GaborApatiNagy) [inline XenAdmin/Commands/ShutDownHostCommand.cs]: The Conditional/Ternary operator returns one of two expressions depending on the condition; in this case, the returned value would always be the same 'sb' object whereas the side-effect would be what we were originally aiming for: to append text to the StringBuilder.

Note that there is no need to return the 'sb', especially to assign it again. Therefore I suggest to have the ternary operator removed from here and have an if-else statement implemented instead.

File: XenAdmin/Commands/ShutDownHostCommand.cs [hunk 2]
```diff
@@ -143,21 +145,40 @@ protected override string ConfirmationDialogText
...
+                        ? sb.AppendFormat(Messages.CONFIRM_SHUTDOWN_HCI_WARN_SERVER, hciHosts[0].Name())
+                        : sb.AppendFormat(Messages.CONFIRM_SHUTDOWN_HCI_WARN_SERVERS, string.Join("\n", hciHosts.Select(h => h.Name())));
+
+                else if (hasRunningVMs)
+                    sb = hosts.Count == 1
```
Comment (GaborApatiNagy) [inline XenAdmin/Commands/ShutDownHostCommand.cs]: see comment on line 156

File: XenAdmin/Commands/ShutDownHostCommand.cs [hunk 3]
```diff
@@ -143,21 +145,40 @@ protected override string ConfirmationDialogText
...
+                        ? sb.AppendFormat(Messages.CONFIRM_SHUTDOWN_SERVER, hosts[0].Name())
+                        : sb.Append(Messages.CONFIRM_SHUTDOWN_SERVERS);
+
+                else sb = hosts.Count == 1
```
Comment (GaborApatiNagy) [inline XenAdmin/Commands/ShutDownHostCommand.cs]: see comment on line 156

File: XenAdmin/Commands/ShutDownHostCommand.cs (hunk 1 above)
Comment (MihaelaStoica) [reply XenAdmin/Commands/ShutDownHostCommand.cs]: Or we can assign the message to a string variable (and continue to use the ternary operator, avoiding lots of ifs) and append it to the StringBuilder further down, before line 169.

File: XenAdmin/Commands/ShutDownHostCommand.cs [hunk 4]
```diff
@@ -143,21 +145,44 @@ protected override string ConfirmationDialogText
...
-                if (hasRunningVMs)
-                    return hosts.Count == 1
-                        ? string.Format(Messages.CONFIRM_SHUTDOWN_SERVER, hosts[0].Name())
-                        : Messages.CONFIRM_SHUTDOWN_SERVERS;
-
-                return hosts.Count == 1
-                    ? string.Format(Messages.CONFIRM_SHUTDOWN_SERVER_NO_VMS, hosts[0].Name())
-                    : Messages.CONFIRM_SHUTDOWN_SERVERS_NO_VMS;
+                    firstWarning = hciHosts.Count == 1
+                        ? firstWarning = string.Format(Messages.CONFIRM_SHUTDOWN_HCI_WARN_SERVER, hciHosts[0].Name())
```
Comment (GaborApatiNagy) [inline XenAdmin/Commands/ShutDownHostCommand.cs]: "firstWarning = " assignments in line 159 and 160 are not needed the value is assigned in line 158

File: XenAdmin/Commands/ShutDownHostCommand.cs [hunk 5]
```diff
@@ -143,21 +145,44 @@ protected override string ConfirmationDialogText
...
+                        : firstWarning = string.Format(Messages.CONFIRM_SHUTDOWN_HCI_WARN_SERVERS, string.Join("\n", hciHosts.Select(h => h.Name())));
+
+                else if (hasRunningVMs)
+                    firstWarning = hosts.Count == 1
+                        ? firstWarning = string.Format(Messages.CONFIRM_SHUTDOWN_SERVER, hosts[0].Name())
```
Comment (GaborApatiNagy) [inline XenAdmin/Commands/ShutDownHostCommand.cs]: same comment as for line 159 and 160

File: XenAdmin/Commands/ShutDownHostCommand.cs [hunk 6]
```diff
@@ -143,21 +145,44 @@ protected override string ConfirmationDialogText
...
+                        : firstWarning = Messages.CONFIRM_SHUTDOWN_SERVERS;
+
+                else firstWarning = hosts.Count == 1
+                    ? firstWarning = string.Format(Messages.CONFIRM_SHUTDOWN_SERVER_NO_VMS, hosts[0].Name())
```
Comment (GaborApatiNagy) [inline XenAdmin/Commands/ShutDownHostCommand.cs]: same comment as for line 159 and 160

**Pull request merged**